distributions/R/scripts/new/header_parser.py

```diff
@@ -0,0 +1,173 @@
+import re
+import argparse
+import ctypes  # used in eval
+import warnings
+from sys import stderr
+from pathlib import Path
+
+from CppHeaderParser import CppHeader, CppMethod, CppVariable
+
+from ctype_translator import mapping
+
+
+param_regex = re.compile(
+    r"""
+    [@\\]param                       # PROPER parameter classifier
+    (?P<mode>(\[(in|out|in,out)])?)  # parameter may be in, out, both or unspecified (def. in)
+    \s*
+    (?P<name>[a-z_][a-z0-9_]*)       # won't match C99's unicode names
+    .*$                              # also capture the description
+    """, re.VERBOSE | re.MULTILINE | re.IGNORECASE
+)
+array_regex = re.compile(r'array\s*of\s*(length|size)\s*(?P<lname>[a-z0-9_]+)')
+
+
+def extract_functions_from_file(path):
+    try:
+        return CppHeader(path).functions
+    except IOError:
+        raise
+    except (TypeError, ValueError, AttributeError) as e:
+        raise ValueError(f'{path} is not a valid C/C++ header file') from e
+
+
+class Parameter:
+    def __init__(self, desc: CppVariable, ordinal: int, mode: str, size: str = None):
+        self.ord = ordinal
+        self.type = mapping[eval(desc['ctypes_type'])]
+        self.name = desc['name']
+        self.mode = mode[1:-1].lower() if mode != '' else 'in'
+        if size is not None:
+            try:
+                if size.startswith('0x'):
+                    self.size = int(size, 16)
+                elif size[0] == '0':
+                    self.size = int(size, 8)
+                else:
+                    self.size = int(size)
+            except ValueError:
+                if size[0].isnumeric():
+                    raise ValueError(f'{size} is neither a variable name nor a numeric literal')
+                else:
+                    self.size = size
+        else:
+            self.size = None
+        self.targets = []  # only used with size parameters
+
+    @property
+    def conversion(self):
+        return f'\t{self.name} <- as.{self.type}({self.name})'
+
+
+def create_wrapper_for_function(fun: CppMethod):
```

Comment:
Add a comment here with examples (!) explaining how typical doxygen comment + function header would be translated into R code. Provide example C code and resulting R code.